### PR TITLE
Adding pigpio

### PR DESCRIPTION
**Pull request by smith-isaac**
Please evaluate build_tarballs.jl for pigpiod_if2 C library and merge.

### CODE REVIEW

File: P/pigpio/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,48 @@
+# Note that this script can accept some limited command-line arguments, run
+# `julia build_tarballs.jl --help` to see a usage message.
+using BinaryBuilder, Pkg
+
+name = "pigpiod_if2"
+version = v"0.0.1"
+
+# Collection of sources required to complete build
+sources = [
+    GitSource("https://github.com/smith-isaac/pigpio.git", "8592e32edaa7d7567b89e648cbf2d53ac6c5ef41")
```
Comment (stemann) [inline P/pigpio/build_tarballs.jl:10]: Why the need for a fork?

File: P/pigpio/build_tarballs.jl [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+]
+
+# Bash recipe for building across all platforms
+script = raw"""
+cd $WORKSPACE/srcdir
+cd pigpio/
```
Comment (giordano) [inline P/pigpio/build_tarballs.jl:16]:
```suggestion
cd $WORKSPACE/srcdir/pigpio/
```

File: P/pigpio/build_tarballs.jl [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    Platform("aarch64", "linux"; libc = "glibc"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "glibc"),
+    Platform("powerpc64le", "linux"; libc = "glibc"),
+    Platform("i686", "linux"; libc = "musl"),
+    Platform("x86_64", "linux"; libc = "musl"),
+    Platform("aarch64", "linux"; libc = "musl"),
+    Platform("armv6l", "linux"; call_abi = "eabihf", libc = "musl"),
+    Platform("armv7l", "linux"; call_abi = "eabihf", libc = "musl")
+]
```
Comment (giordano) [inline P/pigpio/build_tarballs.jl:35]:
```suggestion
platforms = supported_platforms(; exclude=!Sys.islinux)
```

**Review by giordano — COMMENTED** [review]
Some minor improvements, in addition to the question above about why to use a fork

**Comment by smith-isaac** [comment]
Thanks for the recommendations. I had no particular reason for using the fork. I will make a branch and make a pull request there. I will close this pull request.

**Pull request closed without merging**